### DISCUSSION TITLE
過去に実行したbatch runがたまっている状態での初回実行について

### DISCUSSION THREAD

**Discussion by tana-nun** [takeyaqa/magicpod-analyzer#36] [Q&A]
初めまして、早速magicpod-analyzerを使わせていただいています！

出力についてお伺いしたいのですが、magicpod上で過去に実行されたbatch runが複数件溜まっている状態での初回出力はできるものなのでしょうか？
と言いますのも、ローカルでの出力を試していた際、batch runが1件しかないプロジェクトは出力に成功したのですが、複数件溜まっているものはいずれも初回の出力で失敗してしまいました。
2回目last_runの内容で、差分をとっていくということなのでその辺りがうまくいかなかったのかもとかも考えていますが、こういう場合はどういう処理を行えば良いのでしょうか？

**Comment by takeyaqa** [comment]
@tana-nun 返信遅れてしまい申し訳ありません。
初回実行についてですが、「最新の実行から過去100件」を取得するようになっています。なのであまりに古いものでなければ複数件溜まっている状態でも取得はできるようになっているはずです。
エラーの詳細を知りたいので、よければ実行時のコンソールログを共有いただけないでしょうか？　プロジェクト名など公開したくない情報は`***`でマスクしてしまって構いません。ご検討よろしくお願いします。

**Reply by tana-nun** [reply]
@takeya0x86 
お返事ありがとうございます！
溜まっていたのは1年ほど前のものからで300件程度でした。
コンソールログですが以下のようなものでした。
3つのプロジェクトで試してみた時の結果を記しています。

プロジェクト名を日本語から英語にしたりなど色々と条件を変えてみたりしましたが、いずれも以下のようなログでエラーが出ました。
> ***@guestguestnoMacBook-Pro ~ % magicpod-analyzer get-batch-runs --token ***
> INFO  [LocalStore] /Users/***/.magicpod_analyzer/last_run/magicpod.json was successfully loaded. 
> INFO  [MagicPodRunner] Fetching magicpod - ***/*** ... 
> ERROR [MagicPodRunner] Some error raised in '***/***', so it skipped. 
> INFO  [MagicPodRunner] Fetching magicpod - MyOrganization/MyProjectTwo ... 
> ERROR [MagicPodRunner] Some error raised in 'MyOrganization/MyProjectTwo', so it skipped. 
> INFO  [MagicPodRunner] Exporting magicpod workflow reports ... 
> INFO  [LocalExporter] Export test reports to /Users/***/output/20220726-2127-test-magicpod.json 
> INFO  [LocalStore] /Users/***/.magicpod_analyzer/last_run/magicpod.json was successfully loaded. 
> INFO  [LocalStore] /Users/***/.magicpod_analyzer/last_run/magicpod.json was successfully saved. 
> INFO  [MagicPodRunner] Done execute 'magicpod'. status: failure 
> ERROR [LocalExporter]     Error: Some error raised in 'MyOrganization/MyProjectTwo', so it skipped. 


こちらは、一件だけ溜まっている状態のプロジェクトで行い成功したものです。
> ***@guestguestnoMacBook-Pro ~ % magicpod-analyzer get-batch-runs --token ***
> INFO  [LocalStore] /Users/***/.magicpod_analyzer/last_run/magicpod.json was successfully loaded. 
> INFO  [LocalStore] /Users/***/.magicpod_analyzer/last_run/magicpod.json was not found, empty object is used instead. 
> INFO  [MagicPodRunner] Fetching magicpod - ***/*** ... 
> INFO  [MagicPodRunner] Fetching magicpod - MyOrganization/MyProjectTwo ... 
> ERROR [MagicPodRunner] Some error raised in 'MyOrganization/MyProjectTwo', so it skipped. 
> INFO  [MagicPodRunner] Exporting magicpod workflow reports ... 
> INFO  [LocalExporter] Export test reports to /Users/***/output/20220726-2129-test-magicpod.json 
> INFO  [LocalStore] /Users/***/.magicpod_analyzer/last_run/magicpod.json was successfully loaded. 
> INFO  [LocalStore] /Users/***/.magicpod_analyzer/last_run/magicpod.json was not found, empty object is used instead. 
> INFO  [LocalStore] /Users/***/.magicpod_analyzer/last_run/magicpod.json was successfully saved. 
> INFO  [MagicPodRunner] Done execute 'magicpod'. status: failure 
> ERROR [LocalExporter]     Error: Some error raised in 'MyOrganization/MyProjectTwo', so it skipped. 


こちらは10件程度溜まっている別のプロジェクトで失敗したときのものです。
> ***@guestguestnoMacBook-Pro ~ % magicpod-analyzer get-batch-runs -c magicpod_analyzer_cueb.yaml --token=***
> INFO  [LocalStore] /Users/***/.magicpod_analyzer/last_run/magicpod.json was successfully loaded. 
> INFO  [MagicPodRunner] Fetching magicpod - ***/*** ... 
> ERROR [MagicPodRunner] Some error raised in '***/***', so it skipped. 
> INFO  [MagicPodRunner] Fetching magicpod - MyOrganization/MyProjectTwo ... 
> ERROR [MagicPodRunner] Some error raised in 'MyOrganization/MyProjectTwo', so it skipped. 
> INFO  [MagicPodRunner] Exporting magicpod workflow reports ... 
> INFO  [LocalExporter] Export test reports to /Users/***/output/20220726-2131-test-magicpod.json 
> INFO  [LocalStore] /Users/***/.magicpod_analyzer/last_run/magicpod.json was successfully loaded. 
> INFO  [LocalStore] /Users/***/.magicpod_analyzer/last_run/magicpod.json was successfully saved. 
> INFO  [MagicPodRunner] Done execute 'magicpod'. status: failure 
> ERROR [LocalExporter]     Error: Some error raised in 'MyOrganization/MyProjectTwo', so it skipped. 

アドバイスよろしくお願いいたします。
